src/test/java/org/junitpioneer/testkit/assertion/PioneerAssert.java

```diff
@@ -212,4 +264,58 @@ public ExecutionResultAssert hasNumberOfDynamicallyRegisteredContainers(int expe
 				.collect(toList());
 	}
 
+	@SafeVarargs
+	@Override
+	public final ListAssert<String> withExceptionInstancesOf(Class<? extends Throwable>... exceptionTypes) {
+		return assertExceptions(events -> {
+			Stream<Class<? extends Throwable>> classStream = getAllExceptions(events).map(Throwable::getClass);
+			Assertions.assertThat(classStream).containsOnly(exceptionTypes);
```

Comment:
Should it be `only` or `exactly`? (I.e. is it ok if `exceptionTypes` is a superset of actual exceptions?) Whatever makes sense (maybe both?) should probably be reflected in the method name.

Reply:
It's `only`, not `exactly` => `instanceOf(X)` should work like `instanceof` so superset should work.

Reply:
Ok, that makes sense. If we need "exactly", the new method could be called `withExceptionInstancesExactlyOf`.